doc/design/zarr-support.md

```diff
@@ -0,0 +1,100 @@
+The Zarr format is being used by NGFF to support the next generation of microscopy data. This involves a potentially
+nested folder structure with named files inside the folder. In the DANDI this would mean:
+
+1. An asset is associated with a single zarr folder. From a user perspective this is still a single asset and the UI
+   should not try to delve into the structure of the folder. The CLI should be able to download the entire tree. Matt
+   at kitware is looking into IPFS + NGFF, so we should at least keep that in mind.
+3. Blob store allows for a folder, which contains the zarr named "locations" and data. That is given a root prefix,
+   a zarr-compliant software can navigate the zarr metadata/structure using relative path rules.
+3. Blob url should continue to point to a single location (i.e. the prefix ending with `/` common to all keys for zarr files) in an s3 bucket instead of a collection of files.
+4. The `etag` connected with the zarr file should be a tree-hash of dandi-etags of the folder.
```

Comment:
we should decide and point to a specific treehash'ing algorithm here, even if seems obvious -- even little details such as file casing, sorting, etc might play a role here

Reply:
should have read "from the bottom" ;) discussed below https://github.com/dandi/dandi-api/pull/295/files#diff-bd5e6c17b09b2da2d9f2ff15b63630883ecfa993f44f64076dc7cf9ee837095dR29

Reply:
Agreed. I wanted to get buy in on the overall uploading scheme before investing time in the minutae though.